File: scalechain-wallet/src/test/scala/io/scalechain/wallet/CoinAmountSpec.scala
package io.scalechain.wallet

import org.scalatest._


/**
  * Created by kangmo on 5/18/16.
  */
class CoinAmountSpec extends FlatSpec with BeforeAndAfterEach with ShouldMatchers {
  "CoinAmount.from" should "create a CoinAmount with correct value" in {
    CoinAmount.from(100000000) shouldBe CoinAmount( scala.math.BigDecimal(1))
    CoinAmount.from(1) shouldBe CoinAmount( scala.math.BigDecimal(0.00000001))
  }
}

File: scalechain-transaction/src/test/kotlin-not-compiled/io/scalechain/blockchain/transaction/CoinAmountSpec.kt
package io.scalechain.blockchain.transaction

import org.scalatest.*


/**
  * Created by kangmo on 5/18/16.
  */
class CoinAmountSpec : FlatSpec with BeforeAndAfterEach with Matchers {
  "CoinAmount.from" should "create a CoinAmount with correct value" {
    CoinAmount.from(100000000) shouldBe CoinAmount( java.math.BigDecimal(1))
    CoinAmount.from(1) shouldBe CoinAmount( java.math.BigDecimal(0.00000001))
  }
  "coinUnits" should "return a correct units of coins" {
    CoinAmount( java.math.BigDecimal(1)).coinUnits shouldBe 100000000
    CoinAmount( java.math.BigDecimal(0.00000001)).coinUnits shouldBe 1
  }
}
